src/potions.ts

```diff
@@ -0,0 +1,211 @@
+  }
+
+  gross(embezzlers: number, doubleDuration = false) {
+    const bonusMeat = this.bonusMeat();
+    const duration = this.effectDuration() * (doubleDuration ? 2 : 1);
+    // Number of embezzlers this will actually be in effect for.
+    const embezzlersApplied = Math.max(
+      Math.min(duration, embezzlers) - haveEffect(this.effect()),
+      0
+    );
+
+    return (bonusMeat / 100) * (baseMeat * duration + 750 * embezzlersApplied);
+  }
+
+  price(historical: boolean) {
+    // If asked for historical, and age < 14 days, use historical.
+    return historical && historicalAge(this.potion) < 14
+      ? historicalPrice(this.potion)
+      : mallPrice(this.potion);
+  }
+
+  net(embezzlers: number, doubleDuration = false, historical = false) {
+    return this.gross(embezzlers, doubleDuration) - this.price(historical);
+  }
+
+  doublingValue(embezzlers: number, historical = false) {
+    return (
+      Math.max(this.net(embezzlers, true, historical), 0) -
+      Math.max(this.net(embezzlers, false, historical), 0)
+    );
+  }
+
+  useAsValuable(embezzlers: number, doubleDuration = false) {
+    const duration = this.effectDuration() * (doubleDuration ? 2 : 1);
+
+    let quantityToUse = 0;
+    let embezzlersRemaining = Math.max(embezzlers - haveEffect(this.effect()), 0);
+    let keepGoing = true;
+
+    // Use however many will land entirely on embezzler turns.
+    const embezzlerQuantity = Math.floor(embezzlersRemaining / duration);
+    if (this.net(embezzlersRemaining, doubleDuration) > 0 && embezzlerQuantity > 0) {
+      acquire(
+        embezzlerQuantity,
+        this.potion,
+        this.gross(embezzlersRemaining, doubleDuration),
+        false
+      );
+      quantityToUse = Math.max(embezzlerQuantity, itemAmount(this.potion));
+      print(
+        `Determined that ${quantityToUse} ${this.potion.plural} are profitable on embezzlers.`,
+        "blue"
+      );
+      embezzlersRemaining -= quantityToUse * duration;
+    }
+    if (quantityToUse < embezzlerQuantity || (doubleDuration && quantityToUse > 0)) {
+      keepGoing = false;
+    }
+
+    // Now, is there one with both embezzlers and non-embezzlers?
+    if (keepGoing && this.net(embezzlersRemaining, doubleDuration) > 0 && embezzlersRemaining > 0) {
+      acquire(1, this.potion, this.gross(embezzlersRemaining, doubleDuration), false);
+      const additional = Math.max(1, itemAmount(this.potion) - quantityToUse);
+      print(
+        `Determined that ${additional} ${this.potion.plural} are profitable on partial embezzlers.`,
+        "blue"
+      );
+      quantityToUse += additional;
+      embezzlersRemaining = Math.max(embezzlersRemaining - additional * duration, 0);
+    }
+    if (embezzlersRemaining > 0 || (doubleDuration && quantityToUse > 0)) keepGoing = false;
+
+    // How many should we use with non-embezzlers?
+    if (keepGoing && this.net(0, doubleDuration) > 0) {
+      const adventureCap = myAdventures() * 1.04 + 50;
```

Comment:
consider dropping the +50 if not runnning `garbo ascend`

Reply:
Yeah, this is the formula used elsewhere - we should have a global thing that tracks this.